ConnectorActionableMsgs/03 Adaptive Cards/Lab.md

```diff
@@ -86,56 +88,93 @@ This exercise will send the card via email using a console application.
+1. Enter a name for the application. A suggested name is `Expense Card mailer`. Select **Register**.
 
-### Register the application
+1. In the **Overview** blade, copy the **Application (client) ID**.
 
-1. Go to the [Application Registration Portal](https://apps.dev.microsoft.com) and sign in with either a Microsoft account or an Office 365 account.
+1. In the **Overview** blade, , copy the **Directory (tenant) ID**.
 
-1. Select the **Add an app** button. Enter a name for the application and select **Create**.
+1. Select **Authentication** in the left-hand menu.
 
-1. Select the **Add Platform** button and choose **Native Application**.
+1. In the **Redirect URIs** > **Suggested Redirect URIs for public clients (mobile, desktop)** section, select the native client URI. (`https://login.microsoftonline.com/common/oauth2/nativeclient`)
 
-1. Select **Save**.
+    ![Screenshot of application registration showing the Redirect URIs](Images/Exercise2-01.png)
 
-1. Copy the value of **Application ID** for reference later.
+1. Select **Save** from the toolbar at the top of the Authentication blade.
 
-### Add the application ID to the project
+### Compile the SendCardEmail program
 
-1. Open the [App.config](App.config) file.
+1. Launch **Visual Studio 2017**.
+
+1. Open the `SendEmailCard.sln` solution from the **LabFiles\SendCardEmail** folder.
+
+1. Open the **App.config** file in Solution Explorer.
 
 1. Find the following line:
 
     ```xml
     <add key="applicationId" value="[your-app-id-here]" />
+    <add key="tenantId" value="[your-tenant-id-here]" />
     ```
 
-1. Paste the application ID you copied from the portal into the `value`, replacing the token `[your-app-id-here]` and save the file.
+1. Paste the application ID you copied from the portal into the `value`, replacing the token `[your-app-id-here]`.
+
+1. Past the tenant ID you copied from the portal into the `value`, replacing the token `[your-tenant-id-here]`.
 
-### Send card via email
+1. Save the file.
 
-1. Compile and run the **SendAdaptiveCard** solution. The solution is a console application. The application will present an Microsoft Azure Active Directory login prompt. Log in with your credentials. It is not necessary to have any specific permissions. The application will send a mail to the inbox associated to the account used to log in.
+1. Press **Ctrl+Shift+B** in Visual Studio to build the app.
 
-    ![Screenshot of send-adaptivecard-message program.](Images/send-adaptivecard-message-program.png)
+1. An executable program named **SendEmailCard.exe** is compiled into the `bin` folder. This executable is used in the lab.
 
-1. The first time the application is run, you will be asked to consent the application.
+### Send email with MessageCard via console application
```

Comment:
Shouldn't this read "Send email with **Adaptive Card** via console application"?